src/views/HousingLottery/adminView/index.jsx

```diff
@@ -1,23 +1,137 @@
-import { Button, Grid } from '@mui/material';
+import React, { useState, useEffect } from 'react';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  Button,
+  Grid,
+  Typography,
+  Card,
+  CardHeader,
+  CardContent,
+  Link,
+} from '@mui/material';
 import housingService from 'services/housing';
 import styles from '../HousingLottery.module.css';
+import { CSVLink } from 'react-csv';
 
 const AdminView = () => {
-  const handleClick = async () => {
-    const preference = await housingService.getAllPreference();
-    const preferredHall = await housingService.getAllPreferredHall();
-    const applicant = await housingService.getAllApplicant();
-    const schoolYear = await housingService.getAllSchoolYear();
-    console.log('preference' + preference);
-    console.log('preferredHall' + preferredHall);
-    console.log('applicant' + applicant);
-    console.log('schoolYear' + schoolYear);
-  };
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const preference = await housingService.getAllPreference();
+      const preferredHall = await housingService.getAllPreferredHall();
+      const applicants = await housingService.getAllApplicants();
+      const schoolYear = await housingService.getAllSchoolYear();
+
+      // Combine the fetched data into a single array of objects
+      const combinedData = applicants.map((applicant, index) => {
+        const rowData = {
+          lotteryNumber: index + 1,
+          preference: JSON.stringify(preference[index]), // Assuming preference is an object
+        };
+
```

Comment:
Just for clarification, is the variable 'i' here representing each applicant's email address?